packages/xchain-radix/src/client.ts

```diff
@@ -0,0 +1,482 @@
+import { CommittedTransactionInfo, GatewayApiClient } from '@radixdlt/babylon-gateway-api-sdk'
+import {
+  LTSRadixEngineToolkit,
+  NetworkId,
+  PrivateKey,
+  RadixEngineToolkit,
+  SimpleTransactionBuilder,
+} from '@radixdlt/radix-engine-toolkit'
+import {
+  AssetInfo,
+  Balance,
+  BaseXChainClient,
+  FeeOption,
+  FeeType,
+  Fees,
+  Network,
+  PreparedTx,
+  Tx,
+  TxHistoryParams,
+  TxParams,
+  TxType,
+  TxsPage,
+  XChainClientParams,
+} from '@xchainjs/xchain-client'
+import { getSeed } from '@xchainjs/xchain-crypto/lib'
+import { Address, Asset, assetAmount, assetToBase, baseAmount } from '@xchainjs/xchain-util'
+import axios from 'axios'
+import {
+  AssetXRD,
+  KeyType,
+  MAINNET_GATEWAY_URL,
+  RadixChain,
+  STATE_ENTITY_DETAILS_PATH,
+  STOKENET_GATEWAY_URL,
+  TRANSACTION_COMMITTED_DETAILS_PATH,
+  TRANSACTION_CONSTRUCTION_PATH,
+  TRANSACTION_PREVIEW_PATH,
+  TRANSACTION_SUBMIT_PATH,
+  TransferTransactionManifest,
+  XRD_DECIMAL,
+} from './const'
+import { EntityDetailsResponse, Transaction } from './types/radix'
+
+/**
+ * Custom Radix client
+ */
+
+class Client extends BaseXChainClient {
+  privateKey: PrivateKey
+  gatewayApiClient: GatewayApiClient
+  constructor(params: XChainClientParams, keyType: KeyType) {
+    super(RadixChain, { network: params.network, phrase: params.phrase })
+    const seed = getSeed(this.phrase)
```

Comment:
The current implementation of this has some issues:

1. The interface of the constructor implies that we will be deriving a key pair using HDWallets with the `rootDerivationPaths`s present there. However, we do not do that.
2. As David mentioned, idk if we can assume that any hex is a valid private key.
3. If memory serves, the key type is part of our derivation path and therefore it doesn't need to be an argument nor does it require special handling here. It would just require special handling later when deriving the address.

I think that we could/should follow something like what cosmos does in thier SDK and ensure that the derivation paths we use are perhaps the same as the wallet:

https://github.com/radixdlt/xchainjs-lib/blob/e0d9669c549e0899fc30b4251018cebdac6768ca/packages/xchain-cosmos-sdk/src/client.ts#L279-L286

Reply:
got it. I've updated the constructor to use the rootDerivationPaths.
2 should not be a problem any more